libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -90,30 +76,46 @@ @implementation SFSmartSyncSyncManager
 
 #pragma mark - instance access / cleanup
 
-+ (id)sharedInstance:(SFUserAccount *)user {
-    static dispatch_once_t pred;
-    dispatch_once(&pred, ^{
-        syncMgrList = [[NSMutableDictionary alloc] init];
-    });
-    @synchronized([SFSmartSyncSyncManager class]) {
-        if (user) {
-            NSString *key = SFKeyForUserAndScope(user, SFUserAccountScopeCommunity);
-            id syncMgr = [syncMgrList objectForKey:key];
-            if (!syncMgr) {
-                syncMgr = [[SFSmartSyncSyncManager alloc] initWithUser:user];
-                [syncMgrList setObject:syncMgr forKey:key];
-            }
-            return syncMgr;
-        } else {
-            return nil;
++ (void)initialize {
+    if (self == [SFSmartSyncSyncManager class]) {
+        syncMgrList = [NSMutableDictionary new];
+    }
+}
+
++ (instancetype)sharedInstance:(SFUserAccount *)user {
+    if (user) {
+        SFSmartStore *store = [SFSmartStore sharedStoreWithName:kDefaultSmartStoreName user:user];
+        return [self sharedInstanceForStore:store];
+    } else {
+        return nil;
+    }
+}
+
++ (instancetype)sharedInstanceForStore:(SFSmartStore *)store {
```

Comment:
Can we have ```sharedInstance``` and ```sharedInstanceForStore``` pass through a single method underneath? So, this method would call that one with a ```nil``` user, and the previous one would call it with a ```nil``` store, or something along those lines. I'm wondering if there's a possibility that someone would want an instance of a user with a store they pass in.